janusgraph-core/src/main/java/org/janusgraph/graphdb/management/JanusGraphManager.java

```diff
@@ -0,0 +1,199 @@
+
+    private final ConcurrentHashMap<String, Graph> graphs = new ConcurrentHashMap<String, Graph>();
+    private final ConcurrentHashMap<String, TraversalSource> traversalSources = new ConcurrentHashMap<String, TraversalSource>();
+    private Settings settings = null;
+    private final Object instantiateGraphLock = new Object();
+
+    private static JanusGraphManager instance = null;
+
+    public JanusGraphManager(Settings settings) {
+        if (null != this.instance) {
+            String errMsg = "You may not instantiate a JanusGraphManager. The single instance should be handled by Tinkerpop's GremlinServer startup processes.";
+            throw new JanusGraphManagerAlreadyInstantiatedException(errMsg);
+        }
+
+        this.settings = settings;
+        this.instance = this;
+        // Open graphs defined at server start in settings.graphs
+        settings.graphs.forEach((key, value) -> {
+            final StandardJanusGraph graph = (StandardJanusGraph) JanusGraphFactory.open(value, key);
+            if (key.toLowerCase().equals("JanusConfigurationGraph".toLowerCase())) {
+                new ConfigurationManagementGraph(graph);
+            }
+        });
+    }
+
+    public static JanusGraphManager getInstance() {
+        return instance;
+    }
+
+    /**
+     * @Deprecated
+     */
+    @Deprecated
+    public ConcurrentHashMap getGraphs() {
+        return graphs;
+    }
+
+    public Set<String> getGraphNames() {
+        return graphs.keySet();
+    }
+
+    public Graph getGraph(String gName) {
+        return graphs.get(gName);
+    }
+
+    public void putGraph(String gName, Graph g) {
+        graphs.put(gName, g);
+    }
+
+    /**
+     * @Deprecated
+     */
+    public ConcurrentHashMap getTraversalSources() {
+        return traversalSources;
+    }
+
+    public Set<String> getTraversalSourceNames() {
+        return traversalSources.keySet();
+    }
+
+    public TraversalSource getTraversalSource(String tsName) {
+        return null; // JanusGraph do not support storing references to traversalSources
+    }
+
+    public void putTraversalSource(String tsName, TraversalSource ts) {
+        String errMsg = "JanusGraph does not support TraversalSource reference tracking.";
+        throw new JanusGraphManagerDoesNotManageTraversalSourcesException(errMsg);
+    }
+
+    public TraversalSource removeTraversalSource(String tsName) {
+        String errMsg = "JanusGraph does not support TraversalSource reference tracking.";
+        throw new JanusGraphManagerDoesNotManageTraversalSourcesException(errMsg);
+    }
+
+    public Bindings getAsBindings() {
+        final Bindings bindings = new SimpleBindings();
+        graphs.forEach(bindings::put);
+        return bindings;
+    }
+
+    public void rollbackAll() {
+        graphs.forEach((key, value) -> {
+            final Graph graph = value;
+            if (graph.tx().isOpen()) {
+                graph.tx().rollback();
```

Comment:
That graphManager listed above is the exact same graphManager that JanusGraph and Titan used before this changeset, so if that _was_ the case, then it would _still be_ the case.